### PR TITLE
Fix MathJax font override, avoid double initialization

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- maybe roll back some ignored issues on https://github.com/jupyterlab/jupyterlab/pull/14726
- downstream
  - noted on https://github.com/jupyterlite/jupyterlite/issues/1192
  - started on https://github.com/jupyterlite/jupyterlite/pull/1193 by @jtpio 

## Code changes

- [x] make the overload of `TeXFont.defaultFonts` static 
- [x] move inner contents of `protected _ensureInitialized` to a `Private` namespace
  - [x] use a `PromiseDelegate` to avoid race initilization conditions
  - [x] load all used named MathJax imports in parallel
- [x] remove example `404` ignores  
- [x] use `font-display: swap` to reduce FLOUC
  - > tried some `preload`, etc. but the `no-cache` setting makes that... worse, and while loading `Zero` might be a win, it's hard to anticipate the rest...

## User-facing changes

- `.woff` fonts should no longer be downloaded twice from the default URL (`{window.location}/js/output/chtml/`)
- multiple calls to `typeset` should not be able to enter the async init code twice  

## Backwards-incompatible changes

- n/a

### CODE REVIEW

File: packages/mathjax-extension/src/index.ts [hunk 1]
```diff
@@ -157,3 +108,77 @@ const mathJaxPlugin: JupyterFrontEndPlugin<ILatexTypesetter> = {
...
+        import('mathjax-full/js/adaptors/browserAdaptor'),
+        import('mathjax-full/js/a11y/assistive-mml')
+      ]);
+
+      mathjax.handlers.register(
+        AssistiveMmlHandler(SafeHandler(new HTMLHandler(browserAdaptor())))
+      );
+
+      class EmptyFont extends TeXFont {
+        protected static defaultFonts = {} as any;
```
Comment (bollwyvl) [inline packages/mathjax-extension/src/index.ts:152]: this was the actual important thing

**Comment by bollwyvl** [comment]
Presumably the example test passed: certainly not seeing the issue locally any longer. Not sure what the story is on the other UI tests.

**Comment by jtpio** [comment]
Nice, thanks!

I switched the label from `enhancement` to `bug` as this looks more like an issue with the current logic.

Also this can likely be backported to `4.0.x`.

File: examples/example.spec.ts [hunk 2]
```diff
@@ -22,15 +22,7 @@ test('should load the example', async ({ page }) => {
...
-    if (
-      msg.type() === 'error' &&
-      // Ignore 404 for MathJax fonts
-      msg
-        .location()
-        .url.match(
-          /js\/output\/chtml\/fonts\/tex-woff-v2\/MathJax_(Zero|Math-Italic|Main-Regular)\.woff$/
-        ) === null
-    ) {
+    if (msg.type() === 'error') {
```
Comment (jtpio) [inline examples/example.spec.ts:25]: Yeah these ignored error were suggesting something was not correct.

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
@meeseeksdev please backport to 4.0.x

**Comment by bollwyvl** [comment]
I'll take a look at the manual backport when i get a chance.
